### ISSUE TITLE
Whitespace token is_whitespace is False

### ISSUE THREAD

**Issue by SuperBo** [sqlfluff/sqlfluff#847]
I expect segment.is_whitespace of a Whitespace token is True, however, it is set to False.

## Expected Behaviour
segment.is_whitespace return True

## Observed Behaviour
segment.is_whitespace return False
## Version
Include the output of `sqlfluff --version` along with your Python version

## Configuration
```
Include your SQLFluff configuration here
```

**Comment by barrywhart** [comment]
To triage this issue, I searched the SQLFluff code to find all uses of `is_whitespace`. This is the only one I found:
```
src/sqlfluff/core/parser/segments/base.py:72:    is_whitespace = False
```

@alanmcruickshank: What's the purpose of `is_whitespace`?

I see that long ago (2019), there was a class `WhitespaceSegment` (also a `NewlineSegment`). Now it's not a class -- instead, it'd defined in `src/sqlfluff/core/rules/base.py`.
